packages/merge-on-green/test/merge-on-green.ts

```diff
@@ -831,142 +845,76 @@ describe('merge-on-green', () => {
-                author: 'testOwner',
-                reactionId: 1,
-              },
-            ],
-          ];
-          return pr;
-        };
-
-        const scopes = [
-          getPRCleanUp('closed', false),
-          getLabels('automerge'),
-          removeMogLabel('automerge'),
-          removeReaction(),
-        ];
+    describe('adding-a-PR-to-Datastore (addPR) method (no stub)', async () => {
+      let removePRStub: SinonStub;
+      let getPRStub: SinonStub;
 
-        await probot.receive({
-          name: 'schedule.repository' as '*',
-          payload: {org: 'testOwner', cleanUp: true},
-          id: 'abc123',
-        });
-
-        scopes.forEach(s => s.done());
-        assert(removePRStub.called);
+      beforeEach(() => {
+        removePRStub = sandbox.stub(handler, 'removePR');
+        getPRStub = sandbox.stub(handler, 'getPR');
       });
 
-      it('deletes a PR if PR is merged when cleaning up repository', async () => {
-        const scopes = [
-          getPRCleanUp('closed', true),
-          getLabels('automerge'),
-          removeMogLabel('automerge'),
-          removeReaction(),
-        ];
-
-        await probot.receive({
-          name: 'schedule.repository' as '*',
-          payload: {org: 'testOwner', cleanUp: true},
-          id: 'abc123',
-        });
-
-        scopes.forEach(s => s.done());
-        assert(removePRStub.called);
+      afterEach(() => {
+        removePRStub.restore();
+        getPRStub.restore();
       });
+      it('does not add a PR if no branch protection', async () => {
+        //addPRStub.restore();
+        loggerStub.restore();
 
-      it('deletes a PR if label is not found when cleaning up repository', async () => {
         const scopes = [
-          getPRCleanUp('closed', true),
-          getLabels('anotherLabel'),
-          removeMogLabel('automerge'),
-          removeReaction(),
+          // we're purposefully calling an error here
+          getBranchProtection(400, []),
+          commentOnPR(),
         ];
-
+        // eslint-disable-next-line @typescript-eslint/no-var-requires
+        const payload = require(resolve(
+          fixturesPath,
+          'events',
+          'pull_request_labeled'
+        ));
         await probot.receive({
-          name: 'schedule.repository' as '*',
-          payload: {org: 'testOwner', cleanUp: true},
+          name: 'pull_request',
+          payload,
           id: 'abc123',
         });
 
         scopes.forEach(s => s.done());
-        assert(removePRStub.called);
       });
 
-      it('does not delete a PR if it is not invalid', async () => {
-        const scopes = [getPRCleanUp('open', false), getLabels('automerge')];
-
-        await probot.receive({
-          name: 'schedule.repository' as '*',
-          payload: {org: 'testOwner', cleanUp: true},
-          id: 'abc123',
+      it('adds a PR if branch protection when PR labeled', async () => {
+        //addPRStub.restore();
```

Comment:
we can drop this commented out bit I believe. I'd be tempted to use Sinon's sandbox mode, and just always reset the sandbox in a before filter.

Reply:
Oops! Thanks for catching that! I was testing to make sure the sandboxing worked.